skaha/src/main/java/org/opencadc/skaha/registry/ImageRegistryAuth.java

```diff
@@ -0,0 +1,72 @@
+package org.opencadc.skaha.registry;
+
+import ca.nrc.cadc.util.Base64;
+import ca.nrc.cadc.util.StringUtil;
+
+
+/**
+ * Represents credentials to an image registry.  Will be used from the request input as a base64 encoded value.
+ */
+public class ImageRegistryAuth {
+    private static final String VALUE_DELIMITER = ":";
+
+    private final String host;
+    private final String username;
+    private final byte[] secret;
+
+    ImageRegistryAuth(final String username, final byte[] secret, final String host) {
+        if (!StringUtil.hasText(username)) {
+            throw new IllegalArgumentException("username is required.");
+        } else if (secret.length == 0) {
+            throw new IllegalArgumentException("secret value cannot be empty.");
+        } else if (!StringUtil.hasText(host)) {
+            throw new IllegalArgumentException("registry host is required.");
+        }
+
+        this.username = username;
+        this.secret = secret;
+        this.host = host;
+    }
+
+    /**
+     * Constructor to use the Base64 encoded value to obtain the credentials for an Image Registry.
+     *
+     * @param encodedValue The Base64 encoded String.  Never null.
+     * @param host  The registry host for these credentials.
+     * @return ImageRegistryAuth instance.  Never null.
+     */
+    public static ImageRegistryAuth fromEncoded(final String encodedValue, final String host) {
```

Comment:
Yes, in a builder/singleton pattern style since the encoded value won't be preserved, so putting into a constructor didn't seem appropriate:
```java
final ImageRegistryAuth registryAuth = ImageRegistryAuth.fromEncoded(encoded, "docker.io");
```